pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,648 @@
+	if err != nil {
+		return err
+	}
+	if vmexport == nil {
+		return fmt.Errorf("Unable to get '%s/%s' VirtualMachineExport", vmeInfo.Namespace, vmeInfo.Name)
+	}
+
+	// Download the exported volume
+	if err := downloadVolume(client, vmexport, vmeInfo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// downloadVolume handles the process of downloading the requested volume from a VirtualMachineExport
+func downloadVolume(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, vmeInfo *VMExportInfo) error {
+	// Extract the URL from the vmexport
+	downloadUrl, err := GetUrlFromVirtualMachineExport(vmexport, vmeInfo)
+	if err != nil {
+		return err
+	}
+
+	resp, err := handleHTTPRequest(client, vmexport, downloadUrl, vmeInfo.Insecure)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Check server response
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Bad status: %s", resp.Status)
+	}
+
+	output, err := os.Create(vmeInfo.OutputFile)
+	if err != nil {
+		return err
+	}
+	defer util.CloseIOAndCheckErr(output, nil)
+
+	// Lastly, copy the file to the expected output
+	if err := copyFileWithProgressBar(output, resp); err != nil {
+		return err
+	}
+
+	fmt.Println("Download finished succesfully")
+	return nil
+}
+
+// GetUrlFromVirtualMachineExport inspects the VirtualMachineExport status to fetch the extected URL
+func GetUrlFromVirtualMachineExport(vmexport *exportv1.VirtualMachineExport, vmeInfo *VMExportInfo) (string, error) {
+	var downloadUrl string
+
+	if vmexport.Status.Links.External.Volumes == nil {
+		return "", fmt.Errorf("Unable to access the volume info from '%s/%s' VirtualMachineExport", vmexport.Namespace, vmexport.Name)
+	}
+
+	for _, exportVolume := range vmexport.Status.Links.External.Volumes {
+		// Access the requested volume
+		if exportVolume.Name == vmeInfo.VolumeName {
+			for _, format := range exportVolume.Formats {
+				// We always attempt to find and get the compressed file URL
+				if format.Format == exportv1.KubeVirtGz || format.Format == exportv1.ArchiveGz {
+					downloadUrl = format.Url
+					break
+				} else if format.Format == exportv1.KubeVirtRaw {
+					downloadUrl = format.Url
+				}
+			}
+		}
+	}
+
+	if downloadUrl == "" {
+		return "", fmt.Errorf("Unable to get a valid URL from '%s/%s' VirtualMachineExport", vmexport.Namespace, vmexport.Name)
+	}
+
+	return downloadUrl, nil
+}
+
+// waitForVirtualMachineExport waits for the VirtualMachineExport status to be ready
+func waitForVirtualMachineExport(client kubecli.KubevirtClient, vmeInfo *VMExportInfo, interval, timeout time.Duration) error {
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		vmexport, err := getVirtualMachineExport(client, vmeInfo)
+		if err != nil {
+			return true, err
+		}
+
+		if vmexport == nil {
+			return true, err
+		}
+
+		if vmexport.Status == nil {
+			return false, nil
+		}
+
+		done := vmexport.Status.Phase == "Ready"
```

Comment:
This done variable doesn't look needed, you do
```go
if vmexport.Status.Phase != exportv1.Ready {
  fmt.Printf("Waiting for VM Export %s status to be ready...\n", vmeInfo.Name)
  return false, nil
}
...
fmt.Printf("Processing completed successfully\n")
return true, nil
```
If it ever gets to the bottom we know we can return true.